### ISSUE TITLE
Saving doesn't import or remove Go modules

### ISSUE THREAD

**Issue by mislavperi** [zed-industries/community#1247]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Typing in code some struct from a go package and hitting save won't import it. No longer using a package in code (removing `fmt` prints for example) and saving the file will not remove the import from the project.

### Environment

Zed: v0.77.3 (stable)
OS: macOS 13.1.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by JacobWithACapitalJ** [comment]
Just to expand on this, I don't believe the go language server should resolve imports automatically. That instead should be the role of the `goimports` tool.
However, a slightly related bug... when trying to use an external tool to format on save as per the docs:
```go
 "language_overrides": {
        "Go": {
            "format_on_save": {
                "external": {
                    "command": "gofmt",
                    "arguments": [
                        "-r",
                        "{buffer_path}"
                    ]
                }
            }
        }
        ```
        it results in the entire file being wiped and some error logs being created. I'll create a separate issue for this as using `goimports` to format the code will resolve the auto import issue, but appears blocked by a bug

**Comment by aybabtme** [comment]
Similarly in VSCode, it seems like the auto-complete is "aware" of not-yet imported packages, as I can start typing `ioutil.` and the autocomplete will know to look for symbols in `ioutil` even though it's not yet imported in the file. Not sure how they do this, but it's extremely useful.

**Comment by curtisknudson** [comment]
> it results in the entire file being wiped and some error logs being created. I'll create a separate issue for this as using goimports to format the code will resolve the auto import issue, but appears blocked by a bug.

Having this exact same issue. Good to know that it is a known bug.

**Comment by dilutedev** [comment]
, 

> Just to expand on this, I don't believe the go language server should resolve imports automatically. That instead should be the role of the `goimports` tool. However, a slightly related bug... when trying to use an external tool to format on save as per the docs:
> 
> ```json
>  "language_overrides": {
>         "Go": {
>             "format_on_save": {
>                 "external": {
>                     "command": "gofmt",
>                     "arguments": [
>                         "-w",
>                         "{buffer_path}"
>                     ]
>                 }
>             }
>         }
> ```
> 
> it results in the entire file being wiped and some error logs being created. I'll create a separate issue for this as using `goimports` to format the code will resolve the auto import issue, but appears blocked by a bug.

same here hope they resolve it soon.

**Comment by hueys** [comment]
I'm using `goimports` to format and automatically import modules with the following in my settings, and haven't seen any issues so far.

```
    "language_overrides": {
        "Go": {
            "formatter": {
                "external": {
                    "command": "goimports",
                    "arguments": []
                }
            }
        }
    },

```
